src/main/webapp/resources/js/pages/analysis/components/AnalysisProvenance.jsx

```diff
@@ -1,12 +1,206 @@
+  const { analysisOutputsContext, getAnalysisOutputs } = useContext(
+    AnalysisOutputsContext
+  );
+  const [provenance, setProvenance] = useState(null);
+  const [toolInfo, setToolInfo] = useState([]);
+  const [currFileName, setCurrFileName] = useState(null);
+
+  const toolPanelMargin = 5;
+  const levelColors = [
+    {
+      0: blue10,
+      1: blue9,
+      2: blue8,
+      3: blue7,
+      4: blue6,
+      5: blue5,
+      6: blue4,
+      7: blue3,
+      8: blue2
+    }
+  ];
+
+  // Gets the analysis outputs on load if they have not
+  // already been loaded
+  useEffect(() => {
+    if (analysisOutputsContext.outputs === null) {
+      getAnalysisOutputs();
+    }
+  }, []);
+
+  // Get the provenance for the file that the user selected
+  function getProvenance(filename) {
+    if (typeof filename !== "undefined") {
+      if (
+        (provenance !== null &&
+          !provenance.data.filename.includes(filename.toString())) ||
+        provenance === null
+      ) {
+        setCurrFileName(filename.toString());
+        getAnalysisProvenanceByFile(
+          analysisContext.analysis.identifier,
+          filename.toString()
+        ).then(data => {
+          setProvenance(data);
+          setToolInfo([]);
+        });
+      }
+    }
+  }
+
+  // Returns a collapse panel for the tool and it's execution parameters.
+  // Parameters are sorted alphabetically by label
+  function displayToolWithExecutionParameters(tool, margin) {
+    let parameters = [];
+    let sortedParameters = tool.executionParameters.sort((a, b) =>
+      a.parameterName > b.parameterName ? 1 : -1
+    );
+
+    for (let param of sortedParameters) {
+      parameters.push(
+        <Descriptions.Item
+          label={param.parameterName}
+          key={`${tool.toolName}-${param.parameterName}`}
+        >
+          {param.parameterValue}
+        </Descriptions.Item>
+      );
+    }
+
+    return (
+      <Collapse
+        style={{
+          border: "none",
+          borderLeft: `8px ${getLevelColor(margin)} solid`,
+          borderRadius: 0,
+          backgroundColor: grey1
+        }}
+        key={`collapse-${tool.toolName}-${Math.random()}`}
+      >
+        <Panel
+          header={tool.toolName}
+          key={tool.toolName}
+          style={{
+            backgroundColor: grey1,
+            marginLeft: `${3 * margin}px`,
+            borderBottom: "0px",
+            borderRadius: 0
+          }}
+        >
+          {parameters.length > 0 ? (
+            <Descriptions bordered column={2} size="small">
+              {parameters}
+            </Descriptions>
+          ) : (
+            ""
+          )}
+        </Panel>
+      </Collapse>
+    );
+  }
+
+  // Gets the color for the "level" of the execution
+  // tool. Used to display a color gradient along the
+  // left hand side of the provenance for the file to
+  // indicate level
+  function getLevelColor(margin) {
+    let level = margin / toolPanelMargin;
```

Comment:
`level` should be a `const`, it feels funny here to be using this global variable `toolPanelMargin`, maybe pass it in?

Reply:
The reason for using the global variable `toolPanelMargin` is that we want each level to display 5px to the right of the previous level. So I don't think we need to pass it in

Reply:
Changed level to a const in 7165a33809647dafedbd205e730437decb47dde8